runtime/utils/Python.cpp

```diff
@@ -0,0 +1,4 @@
+#include "Python.hpp"
+
+std::mutex python_mutex;
+std::mutex &getPythonMutex() { return python_mutex; }
```

Comment:
Let's also move the `PythonInterpreter` RAII class here.
https://github.com/PennyLaneAI/catalyst/blob/78fd063100dbe84e17482d88c92605de158661ab/runtime/lib/capi/ExecutionContext.hpp#L48

Reply:
https://github.com/PennyLaneAI/catalyst/pull/615/commits/6fd0264a9ab367bf572a5966488ecf2a0d0251a3